datalad/distributed/ora_remote.py

```diff
@@ -851,6 +857,25 @@ def _verify_config(self, gitdir, fail_noid=True):
             self.storage_host, self.store_base_path, self.ria_store_url = \
                 verify_ria_url(self.ria_store_url, url_cfgs)
 
+        self.ria_store_pushurl = self.annex.getconfig('push-url')
+        if self.ria_store_pushurl:
+            if self.ria_store_pushurl.startswith("ria+http"):
+                raise RIARemoteError("Invalid push-url: {}. Pushing over HTTP "
+                                     "not implemented."
+                                     "".format(self.ria_store_pushurl))
+
+            # support URL rewrite without talking to a DataLad ConfigManager
+            # Q is why? Why not use the config manager?
+            url_cfgs = dict()
+            url_cfgs_raw = _get_gitcfg(gitdir, "^url.*", regex=True)
+            if url_cfgs_raw:
+                for line in url_cfgs_raw.splitlines():
+                    k, v = line.split()
+                    url_cfgs[k] = v
```

Comment:
It looks like the code from 867 to here is repeated from above.  Minor and subjective, but perhaps worth extracting into a function.

Reply:
Agree, will do.

Reply:
Didn't move that piece into a function, but no code duplication anymore. Executed unconditionally now and only once (calling `git-config` twice is pointless). The conditional call was about `url` being an alternative to the way older `base-path` and `host` configs, that I'll deprecate in another PR (there are several to come to clean up the mess that is `ora_remote.py`).

Reply:
> calling `git-config` twice is pointless

Ah, true.  Update looks good to me.  Thanks.